### PR TITLE
fix: removed USER directive and added entrypoint to allow volume setup

### LINKED ISSUE #223: Support dynamic volume mounting for log and local storage directories
Currently it's not possible to mount host directory to the Core logs directory: the image is run under `appuser` and `appuser` can't write to the host directories.

This is needed to improve UX for docker-compose files in the [documentation](https://github.com/epam/ai-dial/blob/main/dial-cookbook/docker-compose.yml).

### PR DESCRIPTION
**Pull request by adubovik**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +21,7 @@ FROM eclipse-temurin:17-jdk-alpine
 RUN apk update && apk upgrade --no-cache libcrypto3 libssl3
 # fix CVE-2023-52425
 RUN apk upgrade --no-cache libexpat
+RUN apk add --no-cache su-exec
```
Comment (astsiapanay) [inline Dockerfile:24]: Is there a strong reason using `su-exec` instead of `su`?

Comment (adubovik) [reply Dockerfile:24]: `su-exec` is an idiomatic way to switch users in Docker containers.

Postgres uses it: https://github.com/docker-library/postgres/blob/1424abf76f421d6f7bf933d9e42bbbed866fae3a/12/alpine3.18/docker-entrypoint.sh#L313

Redis uses `gosu`, which is go-implementation of su-exec: https://github.com/docker-library/redis/blob/master/7.2/alpine/docker-entrypoint.sh#L13

`su` is a much more complicated tool. An equivalent would look like this:

```sh
exec su appuser -s /bin/sh -c 'exec "/app/bin/aidial-core" "$@"' _ "$@"
```

which is far from understandable.

**Comment by adubovik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/447328)

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-214.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-214.nightly-test.deltixhub.io
>E2E tests status: success

File: Dockerfile (hunk 1 above)
Comment (astsiapanay) [reply Dockerfile:24]: ok. thx

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-214.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-214.nightly-test.deltixhub.io
>E2E tests status: success

File: Dockerfile [hunk 2]
```diff
@@ -18,22 +21,32 @@ FROM eclipse-temurin:17-jdk-alpine
...
+COPY --chown=appuser:appuser docker-entrypoint.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/docker-entrypoint.sh
 
 HEALTHCHECK --start-period=30s --interval=1m --timeout=3s \
   CMD wget --no-verbose --spider --tries=1 http://localhost:8080/health || exit 1
 
 EXPOSE 8080 9464
-ENTRYPOINT ["/app/bin/aidial-core"]
+
+RUN mkdir -p "$LOG_DIR" && chown -R appuser:appuser "$LOG_DIR"
```
Comment (alexey-ban) [inline Dockerfile]: can we combine RUN commands in one. I also think it’s worth raising them higher that HEALTHCHECK and EXPOSE

Comment (adubovik) [reply Dockerfile]: fixed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-214.nightly-test.deltixhub.io
>E2E tests status: success

**Review by artsiomkorzun — APPROVED** [review]

**Pull request merged**